src/Microsoft.AspNet.OData.Shared/DefaultODataAPIHandler.cs

```diff
@@ -0,0 +1,179 @@
+                {
+                    status = ODataAPIResponseStatus.NotFound;
+                }
+            }
+            catch (Exception ex)
+            {
+                status = ODataAPIResponseStatus.Failure;
+                errorMessage = ex.Message;
+            }
+
+            return status;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryCreate(IDictionary<string, object> keyValues, out TStructuralType createdObject, out string errorMessage)
+        {
+            createdObject = default(TStructuralType);
+            errorMessage = string.Empty;
+
+            try
+            {
+                if (originalList != null)
+                {
+                    originalList = new List<TStructuralType>();
+                }
+
+                createdObject = Activator.CreateInstance(clrType) as TStructuralType;
+                originalList.Add(createdObject);
+
+                return ODataAPIResponseStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return ODataAPIResponseStatus.Failure;
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            try
+            {
+                TStructuralType originalObject = GetFilteredItem(keyValues);
+                originalList.Remove(originalObject);
+
+                return ODataAPIResponseStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return ODataAPIResponseStatus.Failure;
+            }
+        }
+
+        public override IODataAPIHandler GetNestedHandler(TStructuralType parent, string navigationPropertyName)
+        {
+            foreach (PropertyInfo property in clrType.GetProperties())
+            {
+                if (property.Name == navigationPropertyName)
+                {
+                    Type type = typeof(DefaultODataAPIHandler<>).MakeGenericType(property.PropertyType.GetGenericArguments()[0]);
+
+                    return Activator.CreateInstance(type, property.GetValue(parent)) as IODataAPIHandler;
+                }
+            }
+
+            return null;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryAddRelatedObject(TStructuralType resource, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            try
+            {
+                originalList.Add(resource);
```

Comment:
Is there any reason for us to expect this line to throw an exception to justify the `try...catch`?